rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -375,6 +405,34 @@ fn addresses_with_prefix_from_dbus(address_data: &OwnedValue) -> Option<Vec<IpIn
     Some(addresses)
 }
 
+fn routes_from_dbus(route_data: &OwnedValue) -> Option<Vec<IpRoute>> {
+    let route_data = route_data.downcast_ref::<zbus::zvariant::Array>()?;
+    let mut routes: Vec<IpRoute> = vec![];
+    for route in route_data.get() {
+        let route_dict = route.downcast_ref::<zvariant::Dict>()?;
+        let route_map =
+            <HashMap<String, zvariant::Value<'_>>>::try_from(route_dict.clone()).unwrap();
```

Comment:
Instead of potentially panicking I would go for:

```suggestion
            <HashMap<String, zvariant::Value<'_>>>::try_from(route_dict.clone()).ok()?;
```

I know, we do not have a plan B if we cannot get the data, but it looks more polite to me :-)